services/ledger_service.go

```diff
@@ -43,21 +45,12 @@ func (ls LedgerService) QueryDIDDoc(did string) (cheqd.Did, cheqd.Metadata, bool
 		return cheqd.Did{}, cheqd.Metadata{}, false, fmt.Errorf("namespace not supported: %s", namespace)
 	}
 
-	log.Info().Msgf("Connecting to the ledger: %s", serverAddr)
 	conn, err := ls.openGRPCConnection(serverAddr)
 	if err != nil {
 		log.Error().Err(err).Msg("QueryDIDDoc: failed connection")
 		return cheqd.Did{}, cheqd.Metadata{}, false, err
 	}
 
-	defer func(conn *grpc.ClientConn) {
-		err := conn.Close()
-		if err != nil {
-			log.Panic().Err(err).Msg("QueryDIDDoc: failed to close connection")
-			panic(err)
-		}
-	}(conn)
-
```

Comment:
Hmm...

Reply:
Moved to a general method
https://github.com/cheqd/did-resolver/pull/14/files#diff-963627347344b0a90ab9b443cbd3065594b726e6ffa6ac9f46c382d493d97bd3R125